orocos_kdl/src/CMakeLists.txt

```diff
@@ -101,7 +101,7 @@ IF(NOT (CMAKE_VERSION VERSION_LESS 2.8.12))
 ENDIF()
 #####end RPATH
 
-ADD_LIBRARY(orocos-kdl ${LIB_TYPE} ${KDL_SRCS} config.h)
+ADD_LIBRARY(orocos-kdl ${LIB_TYPE} ${KDL_SRCS} "${CMAKE_CURRENT_BINARY_DIR}/config.h")
```

Comment:
is `config.h` needed here at all?

[This](https://github.com/ros2/orocos_kinematics_dynamics/blob/8516c84bedb8afda4332295550ead5752d9d674d/orocos_kdl/src/CMakeLists.txt#L133-L134) and [this](https://github.com/ros2/orocos_kinematics_dynamics/blob/8516c84bedb8afda4332295550ead5752d9d674d/orocos_kdl/src/CMakeLists.txt#L113) line are enough IMO (actually that `set_target_properties()` line is weird, but it doesn't matter for this PR).

Reply:
I think the reason this is here is to ensure that if `config.h.in` was changed, this library would be rebuilt.  But I'm not sure about that.

I think I'm going to go ahead and merge this as-is, as it is the minimal patch to fix this.  We can always come back later on and remove it.

Reply:
I think @ivanpauno's right. Compilers can typically generate files to track header dependencies. I'd be surprised if CMake generators do not do that. The fact it is necessary may suggest there's a different problem lurking.

Having said, this change should be harmless